plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -38,17 +32,11 @@
 public class IcebergSplitSource
         implements ConnectorSplitSource
 {
-    private final CloseableIterable<CombinedScanTask> combinedScanIterable;
     private final Iterator<FileScanTask> fileScanIterator;
 
-    public IcebergSplitSource(CloseableIterable<CombinedScanTask> combinedScanIterable)
+    public IcebergSplitSource(List<FileScanTask> fileScanTasks)
     {
-        this.combinedScanIterable = requireNonNull(combinedScanIterable, "combinedScanIterable is null");
-
-        this.fileScanIterator = Streams.stream(combinedScanIterable)
-                .map(CombinedScanTask::files)
-                .flatMap(Collection::stream)
-                .iterator();
+        this.fileScanIterator = requireNonNull(fileScanTasks, "fileScanTasks is null").iterator();
```

Comment:
@phd3, it looks odd to me that the code here was previously getting all of the file scan tasks, which basically defeats the purpose of using the combined task iterable. Can you give me some background on what's going on here?

That said, I think that this is where the check would go in the older code:

```java
        this.fileScanIterator = Streams.stream(combinedScanIterable)
                .map(CombinedScanTask::files)
                .flatMap(Collection::stream)
                .map(file -> {
                  if (!file.deletes().isEmpty()) {
                    throw new TrinoException(...);
                  }
                  return file;
                }
                .iterator();
```

Reply:
Do you mean we should be assigning the whole `CombinedScanTask` as one split instead of undoing the balancing work by getting individual files from it?

Reply:
In the current implementation, while some `FileScanTasks` can be much smaller, I think the usage of `CombinedScanIterable` helps provide an upper bound on split size, as opposed to the alternative `planFiles`. @electrum may know more if there was a previous discussion around this.

Reply:
(if this is indeed what you're referring to) the # of pending splits are also considered while assigning new splits to Trino's tasks. So I think the imbalance caused by ignoring the CombinedScanTask grouping would be diluted, since the tasks finishing small splits would become eligible for getting assigned some new ones faster.

Initially Iceberg connector's data reader heavily reused hive connector code, so it's likely another reason that we kept using the same model of scanning one part of one file for one Trino split. ( Please ignore this if I'm way off base here from your point : )  )

Reply:
Yes, I think you understood what I was asking. Using the combined tasks means that files will be both split and combined. It seems a little weird to undo combining small tasks into larger ones. You're right that this does provide an upper bound on the split sizes, but I don't know why you would want to remove combining splits across files.

Reply:
Maybe we can move this over to an Issue? It would require changing the Split format I'd rather do it in a separate PR. 

https://github.com/trinodb/trino/issues/8486

Reply:
currently, one `FileScanTask` corresponds to one Trino split, and every Trino task keeps getting assigned more splits based on available capacity. IIUC, assigning a `CombinedScanTask` to a split means that Trino's iceberg reader will need to scan multiple files for every split, which is a bit different from the way things are modeled currently.

Reply:
Okay, so it sounds like Trino handles split combining on its own dynamically. Is that right? If so, then this makes sense. You could also avoid some work by not combining, but it probably doesn't matter much.

Reply:
yes. I'd guess that using combined tasks wouldn't provide a lot of improvement over the current approach.


> You could also avoid some work by not combining

I didn't see any `TableScan` method though that avoids combining, but still divides huge files. I guess we'd need to do that in Trino after using `planFiles`.